Line of sight with callbacks

xZise opened the pull request:
Add methods to allow a callback in the line of sight. So instead of define a transparent byte set, this methods will call a callback on every iteration. Their the callback can define, if this block is “transparent”.

I didn't removed the old line of sight methods, to prevent API breakage, and because I think the classic way is faster.

Fabian

→ [CraftBukkit Pull Request](https://github.com/Bukkit/CraftBukkit/pull/290)

xZise committed:
+callback line of sight methods;

akrieger commented:
It bothers me that the type parameterization on Callback is Result then Parameter, and not Parameter then Result, for some reason.

xZise commented:
Is there any recommendation in which order the generic parameters have to be? And the parameters have speakable names so, using a IDE will show you what the generic definitions are for.

Fabian

akrieger commented:
Sure, but the type theorist in me / programmer in me likes to see things as input -> output, or (args) -> (result). My IDE is vim, so I don't know what you mean by "show what the generic definitions are for" in vim.

xZise commented:
When you create a instance of Callback and just completed ‘Callback’ with the code completion it will complete it to: ‘Callback<Result, Parameter>’.

Also if you have a method you have to check, which parameter comes first. And in Java you first define the result type and then the parameter type(s). Also it is common to assign the content of the right to something on the left (`x := y / 2`, `int x = y / 2`, `add $1, $2, $3`)

Fabian

akrieger commented:
I never said it was wrong, I was just offering my own personal opinion. I could even make a case for your side by saying if you wanted to expand it to have more args, you could just tack them onto the end, as opposed to between the last arg and the result. It's just a preference thing, if you don't agree then you don't have to defend yourself to me.

EvilSeph commented:
I've been meaning to pull this, however it has @author tags which we don't want in our code as they are entirely useless. However, if they are there, I feel that I need to ask you (for permission) to remove them. Please update this pull request for 1.8.1 and remove the @author tags if you want this pulled.

xZise commented:
Okay, I removed all author tags (hopefully) and updated for 1.8.1. I couldn't test it, because one of your [files](https://github.com/Bukkit/CraftBukkit/blob/master/src/main/java/net/minecraft/server/ServerConfigurationManager.java#L321) is “broken” and I don't know where the comment end is intended.

Fabian

xZise commented:
Okay here are the newest files without any author tags (missed the in Callback). I also renamed all methods to avoid incompatibilities with the existing methods where the second parameter was `null`.

Fabian

Dinnerbone commented:
I'm currently cleaning old Pull Requests, and this one will be closed without being accepted.

I'm very sorry for the state that our Pull Requests have come to, and I hope you can all forgive us for letting them sit for so long. We're so undermanned for this, it's unbelievable. We hope that our new Bleeding system will allow for newer pull requests to be picked up and handled by much more people than just ourselves, which will speed up things tremendously.

If you wish to reopen your pull request, please make sure that it's 100% up to date with the current head, and feel free to resubmit whereupon one of our team will properly review it and give it the attention it deserves.

Again, I'm very sorry for letting these pulls sit for so long. Thanks for your patience.

xZise commented:
Okay, I opened a [leaky issue](https://bukkit.atlassian.net/browse/BUKKIT-821) and updated the source code to the current head.

Fabian
